pallets/crowdloans/src/lib.rs

```diff
@@ -531,6 +554,78 @@ pub mod pallet {
     }
 
     impl<T: Config> Pallet<T> {
+        fn distribute_project_tokens(_block_number: T::BlockNumber) -> Result<(), Error<T>> {
+            let mut lock = StorageLock::<Time>::with_deadline(
+                b"distribute_project_tokens::lock",
+                Duration::from_millis(T::LockPeriod::get()),
+            );
+            if lock.try_lock().is_err() {
+                return Err(Error::<T>::GetLockFailed);
+            }
+            // AccountId, cTokenId, percentOfIssuance
+            let ownership_snapshot = Self::ctoken_ownership_snapshot();
+
+            // now for each account lets payout
+            for (account_id, accounts_ownership) in ownership_snapshot.iter() {
+                // payout for each vault
+                for (ctoken, percent_of_ctokens) in accounts_ownership.iter() {
+                    // get the total number of ptokens to be distributed
+                    let ptoken = Self::get_ptoken_from_ctoken(ctoken);
+                    let current_distribution_total = Self::get_distribution_amount(ptoken);
+
+                    // the user should recieve this portion of ptokens
+                    let amount = current_distribution_total * percent_of_ctokens;
+
+                    // transfer tokens to user from this account
+                    T::Assets::transfer(ptoken, &Self::account_id(), &account_id, amount, true)
```

Comment:
we should mint project tokens directly on our chain. Because the tokens are on another chain (recorded on parachain account - 2012/2085), so we can mint it directly from our chain.

Reply:
This needs to be tested and validated, I'll test it